service/season_generator.py

```diff
@@ -40,6 +43,114 @@ async def check_season_end(self):
     async def generate_season(self):
         self._logger.info("Generating new season...")
         async with self._db.acquire() as conn:
-            with open("service/generate_season.sql") as file:
-                query = text(file.read())
-                await conn.execute(query)
+            sql = select([league]).where(league.c.enabled == True)
+            result = await conn.execute(sql)
+            rows = await result.fetchall()
+
+            next_month = datetime.now() + relativedelta(months=1)
+            # season starts and ends at noon, so that all timezones see the same date in the client
+            start_date = datetime(year=next_month.year, month=next_month.month, day=1, hour=12)
+            end_date = start_date + relativedelta(months=SEASON_LENGTH_MONTHS) - relativedelta(days=1)
+
+            for row in rows:
+                await self.update_db(conn, row, start_date, end_date)
+
+    async def update_db(self, conn, league_row, start_date, end_date):
+        season_sql = (
+            select([league_season])
+            .where(league_season.c.league_id == league_row[league.c.id])
+            .order_by(desc(league_season.c.season_number))
+            .limit(1)
+        )
+        result = await conn.execute(season_sql)
+        season_row = await result.fetchone()
+        if season_row is None:
+            self._logger.warning(
+                "No season found for league %s. Skipping this league",
+                league_row[league.c.technical_name]
+            )
+            return
+        result = await conn.execute(select([func.max(league_season.c.id)]))
+        season_id = await result.scalar() + 1
+        season_number = season_row[league_season.c.season_number] + 1
+        season_insert_sql = (
+            insert(league_season)
+            .values(
+                id=season_id,
+                league_id=season_row[league_season.c.league_id],
+                leaderboard_id=season_row[league_season.c.leaderboard_id],
+                placement_games=season_row[league_season.c.placement_games],
+                season_number=season_number,
+                name_key=season_row[league_season.c.name_key],
+                start_date=start_date,
+                end_date=end_date,
+            )
+        )
+        await conn.execute(season_insert_sql)
+
+        division_sql = (
+            select([league_season_division])
+            .where(league_season_division.c.league_season_id == season_row[league_season.c.id])
+        )
+        result = await conn.execute(division_sql)
+        season_division_rows = await result.fetchall()
+        if len(season_division_rows) == 0:
```

Comment:
The pythonic way to check for an empty list is like this.
```suggestion
        if not season_division_rows:
```